usagov/usagov-2021
# Usagov 1717 life event ref langcode

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-1717

## Description

Filters the life events shown in checkboxes to match the language of a benefits category term.
Adds a validator to prevent content editors from referencing a life event from a different language than the term.
For the basic page validator, updates the error message shown to list the benefit categories that don't match the page's language.

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions


A recent db snapshot should be sufficient, since this changes how node add and node edit forms behave.

### Change Requirements



### Validation Steps

- Edit an English benefit category and relate it to life event. You should only see life events that are also English.

![image](https://github.com/user-attachments/assets/7f42a2c6-308f-438e-b6a9-4a8ef045696b)

- Click "Show all life events" and then select one ore more Spanish life events. When you submit the form you should see an error message listing the Spanish life events.

![image](https://github.com/user-attachments/assets/53fca456-601c-4f25-9d07-487eba7ad790)

- Can repeat above for a Spanish benefit category and submitting English life events.

- When editing a basic page in English, click "Show all categories" and select one or more Spanish categories. After you submit the form, the error message displayed should list the Spanish categories.

![image](https://github.com/user-attachments/assets/d0f89f1a-1711-4f9a-aeac-8918433a4511)


## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
